one-line-cr-bot.sh

```diff
@@ -0,0 +1,531 @@
+#!/bin/bash
+#
+# Copyright (C) 2020 Amazon.com, Inc. or its affiliates.
+# Author: Norbert Manthey <nmanthey@amazon.de>
+#
+# Script to report errors/warnings/static code analysis output for a git commit series
+#
+# Tools currently used: infer, cppcheck
+# To potentially add: gcc-10 fanalyzer, cbmc
+#
+# If cppcheck or infer are not present in PATH, the tools can be obtained, and
+# build from their respective source in the www, all in a temporary folder that
+# will be thrown away after the invocation. If you plan to use this script
+# multiple time, consider installing these tools natively! To use this feature,
+# use the flag -I, or assign the value "true" to the environment variable
+# INSTALL_MISSING. Note, installation from the www might fail due to
+# unavailability.
+#
+# To customize the analysis of the used tools, check the documentation for the
+# backends of the one-line-scan tool. Configuration via environment variables is
+# possible.
+
+# Variables that can be influenced via environment, or CLI
+BASE_COMMIT="${BASE_COMMIT:-}"
+BUILD_COMMAND="${BUILD_COMMAND:-make -B}"
+CLEAN_COMMAND="${CLEAN_COMMAND:-make clean}"
+DEBUG="${DEBUG:-}"
+INFER_VERSION="${INFER_VERSION:-1.0.0}" # currently, we only support v1.0.0
+INGORE_ERRORS="${INGORE_ERRORS:-false}"
+INSTALL_MISSING="${INSTALL_MISSING:-false}"
+ONELINESCAN_PARAMS="${ONELINESCAN_PARAMS:-}"
+OUTPUT_FILE="${OUTPUT_FILE:-}"
+OVERRIDE_ANALYSIS_ERROR="${OVERRIDE_ANALYSIS_ERROR:-false}"
+REPORT_NEW_ONLY="${REPORT_NEW_ONLY:-false}"
+WORK_COMMIT="${WORK_COMMIT:-}"
+declare -i VERBOSE
+VERBOSE=${VERBOSE:-0}
+
+# Track tools to be run
+declare -a SUPPORTED_TOOLS=("infer" "cppcheck")
+declare -A RUN_TOOL
+RUN_TOOL["INFER"]="true"
+RUN_TOOL["CPPCHECK"]="true"
+
+#
+# DO NOT TOUCH THE BELOW
+#
+readonly PROG=$(basename "$0")
+readonly SCRIPT_DIR=$(readlink -f "$(dirname "$0")")
+readonly REPOSITORY_DIR=${PWD}
+
+TMP_DATA_FOLDER=""
+
+usage() {
+    cat <<EOF
+One-Line-CR Bot
+
+This script allows to run code analysis based on One-Line-Scan against a given
+code base, and eventually print the diff in defects wrt a base commit. The
```

Comment:
"eventually" makes it sound like this could take forever. How about "code base. Using these results, the user can choose to print the diff in defects in comparison to those found for a base commit."